### PR TITLE
daml-lf: freeze version 1.7

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
fixes: #2289 

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [X] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/java/codegen/BUILD.bazel [hunk 1]
```diff
@@ -21,7 +21,7 @@ load(
 load(
     "//daml-lf/language:daml-lf.bzl",
     "lf_dev_version",
```
Comment (remyhaemmerle-da) [inline language-support/java/codegen/BUILD.bazel]:
```suggestion
    "lf_stable_version",
```

File: language-support/java/codegen/BUILD.bazel [hunk 2]
```diff
@@ -21,7 +21,7 @@ load(
...
-    "lf_latest_version",
+    "lf_stable_version",
```
Comment (remyhaemmerle-da) [inline language-support/java/codegen/BUILD.bazel]:
```suggestion
    "lf_latest_version",
```

File: ledger/test-common/BUILD.bazel [hunk 3]
```diff
@@ -15,11 +15,9 @@ load(
 )
 load(
     "//daml-lf/language:daml-lf.bzl",
-    "lf_latest_version",
+    "lf_stable_version",
```
Comment (remyhaemmerle-da) [inline ledger/test-common/BUILD.bazel]:
```suggestion
    "lf_stable_version",
    "lf_latest_version",
    "lf_dev_version",
```

File: daml-lf/spec/daml-lf-1.rst [hunk 4]
```diff
@@ -247,6 +243,13 @@ Version: 1.dev
 
   * **Add** ``type_rep``, a value presenting a type.
 
+Version: 1.dev
+..............
+
+  * **Change** Transaction submitter must be in the contract key
```
Comment (hurryabit) [inline daml-lf/spec/daml-lf-1.rst]: When did we decide to do this? I'm not convinced requiring the submitter of a `fetchByKey` to be a maintainer of the key is a good idea. This definitely hurts the compositionality of workflows. I also don't see the technical necessity of this change. Being a stakeholder of the contract should absolutely be enough. If you know a contract with the right key exists, then it must be the unique contract with this key. If you don't see a contract with such a key, you can't submit the transaction anyway. (I'm not talking about `lookupByKey` here!)

cc @bame-da @gerolf-da

Comment (hurryabit) [reply daml-lf/spec/daml-lf-1.rst]: In fact, this restriction is pretty much against the spirit of https://github.com/digital-asset/daml/issues/2311, which we all seem to have agreed on.

Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-1.rst]: This is the last work from francesco #1866 already referenced in #2311. 
I just let in dev.

Comment (remyhaemmerle-da) [reply daml-lf/spec/daml-lf-1.rst]: just moving it down from line 227/232 to line 246/252

**Pull request merged**